x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/hyper_parameters.tsx

```diff
@@ -203,6 +216,215 @@ export const HyperParameters: FC<Props> = ({ actions, state, advancedParamErrors
+            value={getNumberValue(alpha)}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFormRow
+          label={i18n.translate('xpack.ml.dataframe.analytics.create.downsampleFactorLabel', {
+            defaultMessage: 'Downsample factor',
+          })}
+          helpText={i18n.translate('xpack.ml.dataframe.analytics.create.downsampleFactorText', {
+            defaultMessage:
+              'Controls the fraction of data that is used to compute the derivatives of the loss function for tree training. Must be between 0 and 1.',
+          })}
+          isInvalid={advancedParamErrors[ANALYSIS_ADVANCED_FIELDS.DOWNSAMPLE_FACTOR] !== undefined}
+          error={advancedParamErrors[ANALYSIS_ADVANCED_FIELDS.DOWNSAMPLE_FACTOR]}
+        >
+          <EuiFieldNumber
+            aria-label={i18n.translate(
+              'xpack.ml.dataframe.analytics.create.downsampleFactorInputAriaLabel',
+              {
+                defaultMessage:
+                  'Controls the fraction of data that is used to compute the derivatives of the loss function for tree training',
+              }
+            )}
+            data-test-subj="mlAnalyticsCreateJobWizardDownsampleFactorInput"
+            onChange={(e) =>
+              setFormState({
+                downsampleFactor: e.target.value === '' ? undefined : +e.target.value,
+              })
+            }
+            step={0.001}
+            min={0}
+            max={1}
+            value={getNumberValue(downsampleFactor)}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFormRow
+          label={i18n.translate('xpack.ml.dataframe.analytics.create.etaGrowthRatePerTreeLabel', {
+            defaultMessage: 'Eta growth rate per tree',
+          })}
+          helpText={i18n.translate('xpack.ml.dataframe.analytics.create.etaGrowthRatePerTreeText', {
+            defaultMessage:
+              'Specifies the rate at which eta increases for each new tree that is added to the forest. Must be between 0.5 and 2.',
+          })}
+          isInvalid={
+            advancedParamErrors[ANALYSIS_ADVANCED_FIELDS.ETA_GROWTH_RATE_PER_TREE] !== undefined
+          }
+          error={advancedParamErrors[ANALYSIS_ADVANCED_FIELDS.ETA_GROWTH_RATE_PER_TREE]}
+        >
+          <EuiFieldNumber
+            aria-label={i18n.translate(
+              'xpack.ml.dataframe.analytics.create.etaGrowthRatePerTreeInputAriaLabel',
+              {
+                defaultMessage:
+                  'Specifies the rate at which eta increases for each new tree that is added to the forest.',
+              }
+            )}
+            data-test-subj="mlAnalyticsCreateJobWizardEtaGrowthRatePerTreeInput"
+            onChange={(e) =>
+              setFormState({
+                etaGrowthRatePerTree: e.target.value === '' ? undefined : +e.target.value,
+              })
+            }
+            step={0.001}
+            min={0.5}
```

Comment:
Just to confirm - we don't currently aim to validate any of the hyperparameter values inside the JSON editor do we?

![image](https://user-images.githubusercontent.com/7405507/107507840-0bacb100-6b98-11eb-8a71-e595dab5e1df.png)

Reply:
We use the 'explain' api to validate these parameters and display any messages returned from that api regarding the particular hyperparameter. There is input validation for value type (e.g. numbers) and min/max, as well.